apps/condo/domains/news/schema/NewsItem.js

```diff
@@ -136,10 +137,40 @@ const NewsItem = new GQLListSchema('NewsItem', {
         },
 
         sendAt: {
-            schemaDoc: 'UTC (!) Date to publish the news item and to send notifications',
+            schemaDoc: 'UTC (!)' +
+                '\nDate to publish the news item and to send notifications.' +
+                '\nIf left blank, it will be published immediately.',
             type: 'DateTimeUtc',
         },
 
+        deliverAt: {
+            schemaDoc: '(Internal auto-calculated field)' +
+                '\nStart time for sending notifications.' +
+                '\nThis field is updated when the "isPublished" field is updated to "true".' +
+                '\nDepends on the "sendAt" field:' +
+                '\n 1) If "sendAt" is empty, then current time + delay of 15 seconds is specified;' +
+                '\n 2) If "sendAt" is specified, then the value is taken from it.',
+            type: 'DateTimeUtc',
+            hooks: {
+                resolveInput: async ({ existingItem, resolvedData, fieldName }) => {
+                    const newItem = { ...existingItem, ...resolvedData }
+                    const prevIsPublished = get(existingItem, 'isPublished', false)
+                    const sendAt = get(newItem, 'sendAt', null)
+                    const isPublished = get(newItem, 'isPublished', false)
+                    const updatedIsPublished = prevIsPublished !== isPublished
+
+                    if (!isPublished || !updatedIsPublished) return get(resolvedData, fieldName)
```

Comment:
Should we set null if the newsItem was unpublished?

Reply:
Yes, we can

Reply:
Updated